offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectDbConnector.groovy

```diff
@@ -0,0 +1,162 @@
+    this.connectionProvider = connectionProvider
+    this.personDbConnector = personDbConnector
+  }
+  
+  public Project addProjectAndConnectPersonsInUserDB(projectIdentifier, projectApplication) {
+    //collect infos needed for database
+    String projectTitle = projectApplication.getProjectTitle()
+    Customer customer = projectApplication.getCustomer()
+    ProjectManager projectManager = projectApplication.getProjectManager()
+    
+    //fetch needed person ids from database
+    int customerID = personDBConnector.getPersonId(customer)
+    int managerID = personDBConnector.getPersonId(projectManager)
+  
+    Connection connection = connectionProvider.connect()
+    connection.setAutoCommit(false)
+
+    connection.withCloseable {it ->
+      try {
+        int projectID = addProjectToDB(it, projectIdentifier, projectTitle)
+        addPersonToProject(it, projectID, managerID, "Manager")
+        addPersonToProject(it, projectID, customerID, "PI")
+      
+        it.commit()
+        
+      } catch (Exception e) {
+        log.error(e.message)
+        log.error(e.stackTrace.join("\n"))
+        it.rollback()
+        it.close() //is this needed?
+        throw new DatabaseQueryException("Could not add person and project data to user database.")
+      }
+    }
+  return new Project(projectIdentifier, projectTitle, projectApplication.getLinkedOffer())
+ }
+
+  private int isProjectInDB(String projectIdentifier) {
+    log.info("Looking for project " + projectIdentifier + " in the DB");
+    String sql = "SELECT * from projects WHERE openbis_project_identifier = ?";
+    int res = -1;
+    Connection connection = connectionProvider.connect()
+    connection.withCloseable { it ->
+      PreparedStatement statement = it.prepareStatement(sql);
+      statement.setString(1, projectIdentifier);
+      ResultSet rs = statement.executeQuery();
+      if (rs.next()) {
+        res = rs.getInt("id");
+      }
+    }
+    return res;
+  }
+
+  public int addProjectToDB(Connection connection, String projectIdentifier, String projectName) {
+    int exists = isProjectInDB(projectIdentifier);
+    if (exists < 0) {
+      log.info("Trying to add project " + projectIdentifier + " to the person DB");
+      String sql = "INSERT INTO projects (openbis_project_identifier, short_title) VALUES(?, ?)";
+      try (PreparedStatement statement =
+      connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+        statement.setString(1, projectIdentifier);
+        statement.setString(2, projectName);
+        statement.execute();
+        ResultSet rs = statement.getGeneratedKeys();
+        if (rs.next()) {
+          logout(conn);
+          log.info("Successful.");
+          return rs.getInt(1);
+        }
+      } catch (Exception e) {
+        log.error("SQL operation unsuccessful: " + e.getMessage());
+        e.printStackTrace();
+      }
+      return -1;
+    }
+    return exists;
```

Comment:
I suggest to introduce an Exception in the method signature to indicate the client, when a project could not be added.

Reply:
please have another look, I replaced the strange int of the check to boolean and try to handle this and other error cases by throwing errors. of course, the project id still needs to be returned in the successful case